sdk/storage/azure-storage-blob/azure/storage/blob/_shared/policies.py

```diff
@@ -0,0 +1,594 @@
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import base64
+import hashlib
+import re
+import random
+from time import time
+from io import SEEK_SET, UnsupportedOperation
+import logging
+import uuid
+import types
+from wsgiref.handlers import format_date_time
+try:
+    from urllib.parse import (
+        urlparse,
+        parse_qsl,
+        urlunparse,
+        urlencode,
+    )
+except ImportError:
+    from urllib import urlencode
+    from urlparse import (
+        urlparse,
+        parse_qsl,
+        urlunparse,
+    )
+
+from azure.core.pipeline.policies import (
+    HeadersPolicy,
+    SansIOHTTPPolicy,
+    NetworkTraceLoggingPolicy,
+    HTTPPolicy)
+from azure.core.pipeline.policies.base import RequestHistory
+from azure.core.exceptions import AzureError, ServiceRequestError, ServiceResponseError
+
+from .models import LocationMode
+
+try:
+    _unicode_type = unicode
+except NameError:
+    _unicode_type = str
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def encode_base64(data):
+    if isinstance(data, _unicode_type):
+        data = data.encode('utf-8')
+    encoded = base64.b64encode(data)
+    return encoded.decode('utf-8')
+
+
+def is_retry(response, mode):
+    """Is this method/status code retryable? (Based on whitelists and control
+    variables such as the number of total retries to allow, whether to
+    respect the Retry-After header, whether this header is present, and
+    whether the returned status code is on the list of status codes to
+    be retried upon on the presence of the aforementioned header)
+    """
+    status = response.http_response.status_code
+    if 300 <= status < 500:
+        # An exception occured, but in most cases it was expected. Examples could
+        # include a 309 Conflict or 412 Precondition Failed.
+        if status == 404 and mode == LocationMode.SECONDARY:
+            # Response code 404 should be retried if secondary was used.
+            return True
+        if status == 408:
+            # Response code 408 is a timeout and should be retried.
+            return True
+        return False
+    if status >= 500:
+        # Response codes above 500 with the exception of 501 Not Implemented and
+        # 505 Version Not Supported indicate a server issue and should be retried.
+        if status in [501, 505]:
+            return False
+        return True
+    return False
+
+
+class StorageBlobSettings(object):
+
+    def __init__(self, **kwargs):
+        self.max_single_put_size = kwargs.get('max_single_put_size', 64 * 1024 * 1024)
+        self.copy_polling_interval = 15
+
+        # Block blob uploads
+        self.max_block_size = kwargs.get('max_block_size', 4 * 1024 * 1024)
+        self.min_large_block_upload_threshold = kwargs.get('min_large_block_upload_threshold', 4 * 1024 * 1024 + 1)
+        self.use_byte_buffer = False
```

Comment:
especially this one.

Reply:
I wont document this class specifically, as it's internal.
I've opened an issue to add to the client docs all the optional configuration parameters:
https://github.com/Azure/azure-sdk-for-python/issues/6165